packages/mobx/src/core/derivation.ts

```diff
@@ -166,10 +166,12 @@ export function checkIfStateReadsAreAllowed(observable: IObservable) {
  */
 export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T, context: any) {
     const prevAllowStateReads = allowStateReadsStart(true)
-    // pre allocate array allocation + room for variation in deps
-    // array will be trimmed by bindDependencies
     changeDependenciesStateTo0(derivation)
-    derivation.newObserving_ = new Array(derivation.observing_.length + 100)
+    // Preallocate array; will be trimmed by bindDependencies.
+    derivation.newObserving_ = new Array(
+        // Reserve constant space for initial dependencies, dynamic space otherwise.
```

Comment:
Sorry for the nitpick, but can you please add PR ID  (#3833) to the comment?

Reply:
Added via https://github.com/mobxjs/mobx/pull/3833/commits/c399fd21b6724f9e212b187abce30974bf93bd58